pkg/virtual/syncer/schemas/builtin/builtin.go

```diff
@@ -0,0 +1,100 @@
+/*
+Copyright 2022 The KCP Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package builtin
+
+import (
+	corev1 "k8s.io/api/core/v1"
+	apiextensionsv1 "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/kube-openapi/pkg/common"
+	"k8s.io/kubernetes/pkg/api/legacyscheme"
+	"k8s.io/kubernetes/pkg/apis/core/install/genericcontrolplane"
+	generatedopenapi "k8s.io/kubernetes/pkg/generated/openapi"
+
+	apisv1alpha1 "github.com/kcp-dev/kcp/pkg/apis/apis/v1alpha1"
+	"github.com/kcp-dev/kcp/pkg/virtual/framework/internalapis"
+)
+
+// syncerSchemas contains a list of internal APIs that should be exposed for the
+// syncer of any SyncTarget.
+var SyncerSchemas map[apisv1alpha1.GroupResource]*apisv1alpha1.APIResourceSchema
+
+func init() {
+	genericcontrolplane.Install(legacyscheme.Scheme)
+	schemes := []*runtime.Scheme{legacyscheme.Scheme}
```

Comment:
what is this used for? don't like to see the legacyscheme in our code.

Reply:
@sttts this isn’t strictly new in this patch set, just being relocated from the syncer API reconciler. I presume the legacy scheme is being used as the sync target is assumed to be Kubernetes API-compatible. Happy to update this to explicitly add each type in the internal API list below to the scheme.

Reply:
I would recommend doing a follow-up to deal with legacyscheme usage as a whole

Reply:
@ncdc happy to open a tracking issue and tackle 👍🏻

Reply:
@hasheddan yes let's do that please, thanks!